src/handlers/tokens/setTokenAuthority.ts

```diff
@@ -0,0 +1,64 @@
+import { TokenClientType } from '@colony/colony-js';
+
+import { ContractEvent } from '~types';
+import {
+  getCachedTokenClient,
+  fetchColoniesByNativeToken,
+  updateSingleColonyNativeTokenStatuses,
+} from '~utils';
+
+export default async (event: ContractEvent): Promise<void> => {
+  const { contractAddress, signature } = event;
+
+  const client = await getCachedTokenClient(contractAddress);
+
+  if (!client) {
+    throw new Error(
+      `Cannot get token client for ${contractAddress} when handling ${signature}`,
+    );
+  }
```

Comment:
This might cause block-ingestor restart loop until a client is found - after crashing and restarting it will attempt to handle this event again I think

It may be the expected behaviour tho, so just checking

Reply:
Makes sense actually. Removed it and just put in a check to handle the event if the client is found. That will still have the desired effect, while preventing the type of behavior you are describing.